clang/lib/Sema/SemaChecking.cpp

```diff
@@ -3145,7 +3138,7 @@ bool Sema::ParseSVEImmChecks(
   return HasError;
 }
 
-static ArmStreamingType getArmStreamingFnType(const FunctionDecl *FD) {
+Sema::ArmStreamingType Sema::getArmStreamingFnType(const FunctionDecl *FD) {
```

Comment:
I don't think this function is worth changing the build configuration to add a dependence of clang::Sema to clang::CodeGen :)

You can just inline (duplicate) the functionality here directly into your function in `AArch64TargetCodeGenInfo::checkFunctionCallABI` and map the attributes directly to the `llvm::AArch64::SMEAttrs` class.

That way you can also avoid needing the `ArmStreamingType` enum.

Reply:
Done.